apps/hash-ai-worker-py/app/typesystem/data_type.py

```diff
@@ -18,3 +60,82 @@ class DataTypeSchema(OntologyTypeSchema, extra=Extra.allow):
     """
 
     ty: str = Field(..., alias="type")
+
+    def _type(self) -> type[DataType]:
+        match self.ty:
+            case "string":
+                return str
+            case "number":
+                return float
+            case "boolean":
+                return bool
+            case "null":
+                return type(None)
+            case "array":
+                return list[Any]
+            case "object":
+                return dict[str, Any]
+            case _:
+                # TODO: activities should never raise an exception, but instead return
+                #  an error value that the workflow can handle.
+                #  https://app.asana.com/0/0/1204934059777411/f
+                msg = f"Unknown type: {self.ty}"
+                raise ValueError(msg)
+
+    async def create_data_type(
+        self,
+        *,
+        actor_id: UUID,
+    ) -> Annotated[Any, "DataTypeAnnotation"]:
+        """Create an annotated type from this schema."""
+        # Custom data types will require an actor ID to be passed in
+        _actor_id = actor_id
```

Comment:
I don't think relevant to this PR, but I am interested: if we create types in the graph API, how is the `owned_by_id` assigned? is it taken from the schema URL? since it might be different to the `actor_id`

Reply:
I'm sorry, I confused two provenance fields. The `actor_id` will be assigned to the `record_created_by`. The `owned_by_id` does not exist for external types.

Reply:
Understood, thank you.

If at some point this AI worker is able to create types in the graph, I think we will need to allow for providing an `owned_by_id` depending on whether the user is triggering the flow from a personal account or with an org account selected. But we can do that when it has that capability